### DISCUSSION TITLE
[Feature] Extensions

### DISCUSSION THREAD

**Discussion by matrixes** [immich-app/immich#13337] [Feature Request]
### I have searched the existing feature requests to make sure this is not a duplicate request.

- [X] Yes

### The feature

I'm very fond of the direction of Immich and would like to be able to add functionality that suits my personal needs more.

One way could be built in extensibility, which would make it possible to create UI's and DB modifications that support those needs.

For instance, I'm often reviewing food and I take photos of dishes to help me remember what I had and what I thought of it.
Currently, I can use the "Description" field of the photo to add that information, but it would be so much easier/useful if I could tweak the UI (and extend the DB a bit) to make a customised form and presentation for the details I'd like to add.

### Platform

- [X] Server
- [X] Web
- [X] Mobile

**Comment by bo0tzz** [comment]
We'll be making a start for some extensibility through #1673. There are no plans for UI plugins at the moment.
